modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/upgrade/v1_0_0/FeatureFlagUpgradeProcess.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.feature.flag.web.internal.upgrade.v1_0_0;
+
+import com.liferay.portal.kernel.feature.flag.constants.FeatureFlagConstants;
+import com.liferay.portal.kernel.portlet.PortalPreferences;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.PortalPreferencesLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortletKeys;
+import com.liferay.portlet.PortalPreferencesWrapper;
+
+/**
+ * @author Thiago Buarque
+ */
+public class FeatureFlagUpgradeProcess extends UpgradeProcess {
+
+	public FeatureFlagUpgradeProcess(
+		CompanyLocalService companyLocalService,
+		PortalPreferencesLocalService portalPreferencesLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_portalPreferencesLocalService = portalPreferencesLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() {
+		_companyLocalService.forEachCompanyId(
+			companyId -> {
+				PortalPreferencesWrapper portalPreferencesWrapper =
```

Comment:
@thiago-buarqque Another question is, for the portal preferences, and even feature flags, if you have that activate logic in the `CompanyModelListener`, is this upgrade necessary? Could you explain why it's needed for a better understanding?

Reply:
Ops, sorry I should've explained that. I thought Shuyang would review this and he has the context...

This upgrade process ensures that if I created my company at 7.4 u120 and upgraded to 7.4 u130 (or whatever version this is going out) that company will be marked as processed, and new deprecation feature flags will remain as true (default value).

If I do not mark as processed, old companies will be catch by the CompanyModelListener activate method and would disable the feature flags

Reply:
Also, the reason why the activate method exist is because, for a brand new liferay setup, it won't catch the company right after creation, since it's activated after the portal is initialized

Reply:
Thanks! Please make sure you reflect this in the commit message, so in the future when people look at this part they can also know the context.

Reply:
Done!